react/features/base/participants/reducer.js

```diff
@@ -159,6 +164,9 @@ function _participant(state: Object = {}, action) {
  */
 ReducerRegistry.register('features/base/participants', (state = [], action) => {
     switch (action.type) {
+    case CONFERENCE_LEFT:
+        return state.map(p => _participant(p, action));
+
```

Comment:
🤔

* Why `return state.map(p => _participant(p, action));` and the empty line after it? Isn't the `case` exactly the same as the 4 `case`s right after it? I think it's an unnecessary duplication which is in violation of our coding style.

* Why `CONFERENCE_LEFT` and why it only? I don't think we should be going on with `CONFERENCE_LEFT` anymore. We've talked how we often have to handle`CONFERENCE_LEFT` and `CONFERENCE_FAILED`. I think we should look into introducing actions for more general, catch-all events. For example, events similar to the Web's such as `beforeunload` (that'd be something around `SET_LOCATION_URL` I presume).